fenix/app/src/main/java/org/mozilla/fenix/translations/preferences/downloadlanguages/DownloadLanguagesPreference.kt

```diff
@@ -0,0 +1,508 @@
+            style = FirefoxTheme.typography.subtitle1.copy(
+                color = FirefoxTheme.colors.textPrimary,
+            ),
+            linkTextDecoration = TextDecoration.Underline,
+        )
+    }
+}
+
+@Composable
+private fun downloadLanguageItemContentDescriptionPreference(
+    item: DownloadLanguageItemPreference,
+    itemDescription: String,
+): String {
+    val contentDescription: String
+
+    when (item.state.status) {
+        DownloadLanguageItemStatusPreference.Downloaded -> {
+            contentDescription =
+                item.languageModel.language?.localizedDisplayName + " " + itemDescription + stringResource(
+                    id = R.string.download_languages_item_content_description_downloaded_state,
+                )
+        }
+
+        DownloadLanguageItemStatusPreference.NotDownloaded -> {
+            contentDescription =
+                item.languageModel.language?.localizedDisplayName + " " + itemDescription + " " + stringResource(
+                    id = R.string.download_languages_item_content_description_not_downloaded_state,
+                )
+        }
+
+        DownloadLanguageItemStatusPreference.DownloadInProgress -> {
+            contentDescription =
+                item.languageModel.language?.localizedDisplayName + " " + itemDescription + " " + stringResource(
+                    id = R.string.download_languages_item_content_description_in_progress_state,
+                )
+        }
+
+        DownloadLanguageItemStatusPreference.Selected -> {
+            contentDescription =
+                item.languageModel.language?.localizedDisplayName + " " + itemDescription + stringResource(
+                    id = R.string.download_languages_item_content_description_selected_state,
+                )
+        }
+    }
+    return contentDescription
+}
+
+@Composable
+private fun IconDownloadLanguageItemPreference(item: DownloadLanguageItemPreference) {
+    when (item.state.status) {
+        DownloadLanguageItemStatusPreference.Downloaded -> {
+            if (item.state.type != DownloadLanguageItemTypePreference.PivotLanguage) {
+                Icon(
+                    painter = painterResource(
+                        id = R.drawable.ic_delete,
+                    ),
+                    contentDescription = null,
+                    tint = FirefoxTheme.colors.iconPrimary,
+                )
+            }
+        }
+
+        DownloadLanguageItemStatusPreference.NotDownloaded -> {
+            if (item.state.type != DownloadLanguageItemTypePreference.PivotLanguage) {
+                Icon(
+                    painter = painterResource(
+                        id = R.drawable.ic_download,
+                    ),
+                    contentDescription = null,
+                    tint = FirefoxTheme.colors.iconPrimary,
+                )
+            }
+        }
+
+        DownloadLanguageItemStatusPreference.DownloadInProgress -> {
```

Comment:
I think the button state should additionally be switched to 'disabled' while `DownloadInProgress` because it will be a no-op.

Reply:
Done

Reply:
Thanks, could you take another quick look at this? The focus ring is different now - I think the focus ring should remain the same as the others for a download in progress, only just have the button state set to be semantically disabled. 

It is now showing up in TalkBack as a focus ring around the text portion only and identifying as an active button.

Reply:
Done , I fixed it.

Reply:
Perfect, looks great, thanks!